### DISCUSSION TITLE
Why cortex recommend Memcached over Redis

### DISCUSSION THREAD

**Discussion by mengmengy** [cortexproject/cortex#4411] [Q&A] [answered]
Hello,

According to this cortex [documentation](https://cortexmetrics.io/docs/chunks-storage/caching/), Memcached is recommended over Redis, is there any special reasons behind it? Is it because Redis is relatively new added to Cortex and it's not stable enough for production use right now? 
I have read articles about Memcached and Redis, I think both Memcached and Redis have high performance and availability, Redis has pros like persistent data, cluster support etc.  I'm wondering which cache would you recommend in the perspective of cortex if I just want to cache the query results? Appreciate for your opinions!

**Comment by bboreham** [comment]
The people who wrote that page know Memcached better than Redis.
Also note that entire section is deprecated.

**Comment by silveiralexf** [comment]
@bboreham do you know if is there expectation regarding when Redis use would be considered production-ready? Thanks in advance!

**Comment by bboreham** [comment, marked as answer]
No defined timescale.  I think it needs a few people who use Redis to report back.

I also found #3573 where there was some discussion about how to add Redis support for blocks storage but it didn't get done.

**Reply by silveiralexf** [reply]
thanks a ton @bboreham for the quick response!

**Reply by mengmengy** [reply]
Is Reis cache **stable** enough for production use except lacking blocks storage support?

**Reply by bboreham** [reply]
It's stable in the sense that nobody is changing it, but for how well it works we need someone using Redis to answer.
